src/Scientist/Internals/Experiment.cs

```diff
@@ -10,7 +10,25 @@ internal class Experiment<T, TClean> : IExperiment<T, TClean>, IExperimentAsync<
 
         private static readonly Func<Task<bool>> _alwaysRun = () => Task.FromResult(true);
         private static readonly Action<Operation, Exception> _alwaysThrow
-            = (operation, exception) => { throw exception; };
+            = (operation, exception) =>
+            {
+                string exceptionMessage;
+                switch (operation)
+                {
+                    case Operation.Clean:
+                    case Operation.Compare:
+                    case Operation.Enabled:
+                    case Operation.Ignore:
+                    case Operation.Publish:
+                    case Operation.RunIf:
+                        exceptionMessage = $"An exception occurred within the experiment during the '{operation.ToString()}' operation.";
+                        break;
+                    default:
+                        exceptionMessage = "An exception occurred within the experiment, the operation was unknown.";
+                        break;
+                }
+                throw new Exception(exceptionMessage, exception);
```

Comment:
Does it make sense to create a new `OperationException` type for this situation?  We could pass the specific `Operation`, `innerException` and then internally handle the rest.  This would allow the caller if necessary to catch and handle the specific `Operation` instead of having to parse the message.